---
layout: post
title: "Unaligned Low-rank Tensor Regression with Attention (ULTRA)"
date: 2024-06-17 12:00:00 -0500
excerpt: "A tensor-learning algorithm I invented for automatically constructing
mechanistic explanations of arbitrary sample metadata (e.g. patient disease
status) from scRNA-seq."
importance: 10
---

I invented and implemented this method as a way to automatically construct
mechanistic explanations of arbitrary sample metadata (e.g. patient disease
status) from single-cell RNA sequencing data.

I discovered that a very natural way to solve this problem is to use a form of
*cell attention* to pick out cell types for which the expression of some gene(s) is maximally
covariant with the response variable.

While existing tensor decomposition methods allow the reduced-rank
regression of regular (i.e. aligned) tensors (e.g.
[CP-PLSR](https://www.semanticscholar.org/paper/Multiway-calibration.-Multilinear-PLS-Bro/5d7a1f4c37d36b804183d4240a304d1539cf5551))
and there exist unsupervised tensor decomposition methods for unaligned
tensors (e.g. PARAFAC2, which [our lab also applied to
scRNA](https://www.biorxiv.org/content/10.1101/2024.07.29.605698v1)), there
is currently no way to perform reduced-rank regression on unaligned tensors,
which is the structure of scRNA-seq data.

This method automatically answers a very typical type of question that
researchers ask in scRNA, of the form "which genetic programs in which subsets
of cells are responsible for this phenotype".

This is a work in progress. For the math and preliminary results, see [my
slides](/assets/mm-presentation.key).
